diff --git a/node_modules/jsfuzz/build/src/versifier.js b/node_modules/jsfuzz/build/src/versifier.js
index d1937e4..d016620 100644
--- a/node_modules/jsfuzz/build/src/versifier.js
+++ b/node_modules/jsfuzz/build/src/versifier.js
@@ -481,6 +481,7 @@ function tokenize(data) {
             res.push(new ControlNode(r));
         }
         i += s;
+        if (res.length > 1000) break;
     }
     if (state === StateType.stateAlpha) {
         res.push(new AlphaNumNode(makeDict(data.slice(start))));
